esp/services/ws_machine/ws_machineService.cpp

```diff
@@ -1189,6 +1189,9 @@ int Cws_machineEx::runCommand(IEspContext& context, const char* sAddress, const
         int len = response.length();
         if (len > 0 && response.charAt(--len) == '\n') // strip newline
           response.setLength(len);
+        if (len > 0 && !strnicmp(response.str(), "stderr:", 7))
```

Comment:
Should you strip off the leading "stderr:" part from the response here ?

Reply:
How about changing it to "Error:"?

Reply:
I think we should figure out why wait() returns 0 when cmd is not found (instead of 127) and if we can do that then use that logic to know if it failed, otherwise assume it ran ok.

Reply:
Yes, if IFRunSSH.exec() can send back 127, I should check that.

Reply:
OK - so we know now it always returns 0 because it runs in the background.  If this runs in a new, separate thread then I think I would remove this code that checks for stderr: and just run it in the foreground (ie set background to false in exec() and then no -f flag on ssh cmdline) and it should automatically handle a return code of not zero.  But test it with a sleep in preflight to make sure that doesn't affect the GUI behavior if it takes a long time to complete.

Reply:
I did the test with a sleep in preflight and it doesn't affect the GUI behavior. I also found that, if I set background to false in exec() and the preflight cannot be found, the exec() will throw an exception with code 127.